pkg/controller/nodeconfig/resource.go

```diff
@@ -242,52 +273,46 @@ done
 
 find /host -mindepth 1 -maxdepth 1 -type l -exec cp -P "{}" ./ \;
 
-# Required for node setup
-for dir in "run/udev" "run/mdadm" "run/dbus"; do
-	if [ -d "/host/${dir}" ]; then
-		mkdir -p "./${dir}"
-		mount --rbind "/host/${dir}" "./${dir}"
-	fi
-done
-
-# Required by CRI client
-for dir in "run/crio" "run/containerd"; do
-	if [ -d "/host/${dir}" ]; then
-		mkdir -p "./${dir}"
-		mount --rbind "/host/${dir}" "./${dir}"
-	fi
-done
-
-if [ -f "/host/run/dockershim.sock" ]; then
-	touch "./run/dockershim.sock"
-	mount --bind "/host/run/dockershim.sock" "./run/dockershim.sock"
-fi
-
-# Required by node tuning
-if [ -d "/host/var/lib/kubelet" ]; then
-	mkdir -p "./var/lib/kubelet"
-	mount --rbind "/host/var/lib/kubelet" "./var/lib/kubelet"
-fi
+# Create /scylla-operator directory for additional files.
+mkdir './scylla-operator'
 
 # Mount operator binary
-mkdir -p ./scylla-operator
-touch ./scylla-operator/scylla-operator
-mount --bind /usr/bin/scylla-operator ./scylla-operator/scylla-operator
+mkdir -p './scylla-operator/usr/bin/'
+touch './scylla-operator/usr/bin/scylla-operator'
+mount --bind {,./scylla-operator}/usr/bin/scylla-operator
 
-# Mount service account
-mkdir -p "./run/secrets/kubernetes.io/serviceaccount"
-for f in ca.crt token; do
-	touch "./run/secrets/kubernetes.io/serviceaccount/${f}"
-	mount --bind "/run/secrets/kubernetes.io/serviceaccount/${f}" "./run/secrets/kubernetes.io/serviceaccount/${f}"
-done
+# Mount container run to propagate secrets and configmaps.
+mkdir './scylla-operator/run'
+mount --rbind {,./scylla-operator}/run
+							
+# Mount container tmp.
+mkdir './scylla-operator/tmp'
+mount --rbind {,./scylla-operator}/tmp
```

Comment:
nit: is this necessary? ./scylla-operator is already in tmp, so wouldn't it actually be cleaner to not bind this?

Reply:
in hosts's tmp, we don't want to write to the host fs